KeenClient/KIOEventStore.m

```diff
@@ -685,6 +689,12 @@ - (void)closeDB {
 }
 
 - (id)convertNSDateToISO8601:(id)date {
+    __block NSString *iso8601 = nil;
```

Comment:
@heitortsergent we should probably reopen it. I'm surprised closing it like we do hasn't come back to cause problems up until this point.

It might make sense to remove most of the `[self closeDB];` calls and instead call it from `handleSQLiteFailure` when a failure occurs. That way we can reopen it after we close it. That or we can open/close on every transaction.

Thoughts?